src/components/Canvas/CanvasTemplate.vue

```diff
@@ -121,83 +149,96 @@ export default {
+      // first loop to analyze each blocks
+      // Second loop to analyze each images in the block
+      for (let i = 0; i <= this.imageBlockController.length; i++) {
+        const currentImageBlock = this.$refs["image-block-" + i];
+        if (currentImageBlock) {
+          const nbElementsInBlock = Object.keys(currentImageBlock.$refs).length;
+          for (let j = 0; j < nbElementsInBlock; j++) {
+            this.imageCollisionAnalyzis(
+              currentCenterLeftPosition,
+              currentCenterTopPosition,
+              currentImageBlock.$refs,
+              "image-element-" + j,
+              this.imageBlockController[i].relatedImages[j],
+              this.imageBlockController[i].imagePositions[j]
+            );
+          }
+        }
+      }
+
+      // Manage size of the focus rectangle
+      this.$refs["rectangle"].hasFocus = this.currentFocus.includes(true);
+    },
+    imageCollisionAnalyzis(
+      currentCenterLeftPosition,
+      currentCenterTopPosition,
+      refToAnalyze,
+      ref,
+      focusElement,
+      imagePosition
+    ) {
+      const imageToAnalyze = refToAnalyze[ref];
+      if (imageToAnalyze) {
+        const imageToAnalyzeLeftPositionWithPixel =
+          imageToAnalyze.position.left;
+        // Remove px chars and convert to number
+        const imageToAnalyzeImageLeftPosition = +imageToAnalyzeLeftPositionWithPixel.substring(
           0,
-          currentImageLeftPositionString.length - 2
+          imageToAnalyzeLeftPositionWithPixel.length - 2
         );
-        const currentImageTopPositionString = this.$refs["image-block"].$refs[
-          "image-element-" + i
-        ].position.top;
-        const currentImageTopPosition = +currentImageTopPositionString.substring(
+        const imageToAnalyzeTopPositionWithPixel = imageToAnalyze.position.top;
+        // Remove px chars and convert to number
+        const imageToAnalyzeImageTopPosition = +imageToAnalyzeTopPositionWithPixel.substring(
           0,
-          currentImageTopPositionString.length - 2
+          imageToAnalyzeTopPositionWithPixel.length - 2
         );
+        const imageId = imageToAnalyze.imageId;
         // Check collision
         this.collisionAnalysis(
+          imageId,
           currentCenterLeftPosition,
           currentCenterTopPosition,
-          currentImageLeftPosition,
-          currentImageTopPosition,
-          this.$refs["image-block"].$refs["image-element-" + i]
+          imageToAnalyzeImageLeftPosition,
+          imageToAnalyzeImageTopPosition,
+          focusElement,
+          imagePosition
         );
       }
-      // Manage size of the focus rectangle
-      this.$refs["rectangle"].hasFocus = this.currentFocus.includes(true);
     },
     collisionAnalysis(
+      imageId,
       currentCenterLeftPosition,
       currentCenterTopPosition,
       currentImageLeftPosition,
       currentImageTopPosition,
```

Comment:
probably change all positions to object with {top, left} so to rename params `currentCenter` and `currentImage`